src/odemis/driver/orsay.py

```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+
+        for aprtr_nmbr in range(NMBR_APERTURES):
+            connector_refs = self._apertureConnectors[f"Aperture{aprtr_nmbr}"]
+            connector_refs.append(OrsayParameterConnector(self._apertureDict[f"Aperture{aprtr_nmbr}"]["LifeTime"],
+                                                          recursive_getattr(self._hybridAperture,
+                                                                            f"Aperture{aprtr_nmbr}.Lifetime"))
+                                  )
+            connector_refs.append(OrsayParameterConnector(self._apertureDict[f"Aperture{aprtr_nmbr}"]["Size"],
+                                                          recursive_getattr(self._hybridAperture,
+                                                                            f"Aperture{aprtr_nmbr}.Size"))
+                                  )
+            connector_refs.append(OrsayParameterConnector(self._apertureDict[f"Aperture{aprtr_nmbr}"]["PositionX"],
+                                                          recursive_getattr(self._hybridAperture,
+                                                                            f"Aperture{aprtr_nmbr}.PositionX"))
+                                  )
+            connector_refs.append(OrsayParameterConnector(self._apertureDict[f"Aperture{aprtr_nmbr}"]["PositionY"],
+                                                          recursive_getattr(self._hybridAperture,
+                                                                            f"Aperture{aprtr_nmbr}.PositionY"))
+                                  )
+
+        self.update_VAs()
+        self._metadata[model.MD_APERTURES_INFO] = self._getAllApertureMetadata()
+
+    def selectAperture(self, id):
+        return self._executor.submit(self._selectAperture, id)
+
+    def _selectAperture(self, id, timeout=10):
+        logging.debug(f"Aperture {id} is selected")
+        tend = time.time() + timeout
+        self._hybridAperture.SelectedDiaph.Target = id
+        while int(self._hybridAperture.SelectedDiaph.Actual) != id \
+                and self._hybridAperture.IsMovingX.Actual == "True" \
+                and self._hybridAperture.IsMovingY.Actual == "True":
+            time.sleep(0.1)
+            t = time.time()
+            if t > tend:
+                raise TimeoutError("Referencing timeout after %g s" % timeout)
+
+        self._selectedApertureConnector.update_VA()
+        self._sizeSelectedApertureConnector.update_VA()
+
+    def _doMoveAbs(self, pos, timeout=10):
+        logging.debug(f"Moving the aperture to position {pos}")
+        tend = time.time() + timeout
+        self._hybridAperture.XPosition.Target = pos["x"]
+        self._hybridAperture.YPosition.Target = pos["y"]
+
+        # TODO K.K. test and check isMovingX/Y with Wilco
+        while not float(self._hybridAperture.XPosition.Actual) != pos \
+                or not float(self._hybridAperture.YPosition.Actual) != pos \
```

Comment:
When you compare floats, especially when converted from string, you have to use `almost_equal`. Here, you might even want to "bump up" the absolute error to a tiny value (eg 1µm), to make sure that if the position is pretty close, that counts.